### PR TITLE
Add Support for Packetflux RackInjector

### PR DESCRIPTION
**Pull request by kvedder-amplex**
Please give a short description what your pull request is for

This PR adds support for PacketFlux Rack Injectors. 

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [X] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/polling/os/packetfluxpfr.inc.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
```
Comment (Jellyfrog) [inline includes/polling/os/packetfluxpfr.inc.php:1]: Please convert this to the new os format, https://docs.librenms.org/Developing/os/Initial-Detection/#os-discovery

File: includes/polling/os/packetfluxpfr.inc.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+// Grab the coordinates read from GPS stats and update the location if different
+
+// PACKETFLUX-GNSS-MIB::gnssLatitude.0.0
+$latitude = snmp_get($device, '.1.3.6.1.4.1.32050.3.4.1.1.3.0.0','-OQv', 'PACKETFLUX-GNSS-MIB');
```
Comment (Jellyfrog) [inline includes/polling/os/packetfluxpfr.inc.php:17]: These snmp_get should be converted to a multi call.

File: includes/discovery/sensors/count/packetfluxpfr.inc.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+#PACKETFLUX-GNSS-MIB::gnss1PPSMissingPulses.0.0
+$oid = '.1.3.6.1.4.1.32050.3.4.1.1.14.0.0';
+$current = (snmp_get($device, $oid, '-Oqv') / 1);
```
Comment (Jellyfrog) [inline includes/discovery/sensors/count/packetfluxpfr.inc.php:5]: These snmp_get should be converted to a multi call.

**Review by Jellyfrog — CHANGES_REQUESTED** [review]

File: includes/polling/os/packetfluxpfr.inc.php (hunk 1 above)
Comment (kvedder-amplex) [reply includes/polling/os/packetfluxpfr.inc.php:1]: Took a run at this. The original code was produced by a co-worker. Let me know if I need to make further changes.

File: includes/polling/os/packetfluxpfr.inc.php (hunk 2 above)
Comment (kvedder-amplex) [reply includes/polling/os/packetfluxpfr.inc.php]: Gave these a whirl.

File: includes/discovery/sensors/count/packetfluxpfr.inc.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+#PACKETFLUX-GNSS-MIB::gnss1PPSMissingPulses.0.0
+$oid = '.1.3.6.1.4.1.32050.3.4.1.1.14.0.0';
+$current = (snmp_getnext_multi($device, $oid, '-Oqv') / 1);
```
Comment (Jellyfrog) [inline includes/discovery/sensors/count/packetfluxpfr.inc.php:5]: Sorry, I meant doing 1 "snmp multi" request per file, so only one request to go the device. Check other OS for inspiration

Comment (kvedder-amplex) [reply includes/discovery/sensors/count/packetfluxpfr.inc.php:5]: Understood. Thanks for the clarification!

File: includes/polling/os/packetfluxpfr.inc.php (hunk 2 above)
Comment (kvedder-amplex) [reply includes/polling/os/packetfluxpfr.inc.php]: I reworked the logic here, two of the calls could be combined, The third uses a different MIB so from what I can gather it needs to be separate. My PHP is rusty, would I be on the right track here?

File: includes/polling/os/packetfluxpfr.inc.php (hunk 1 above)
Comment (kvedder-amplex) [reply includes/polling/os/packetfluxpfr.inc.php:1]: So for this I would need to add a file to cover device discovery as referenced in the above link. The current code doesn't do discovery, and adding that feature is what is being requested?

**Comment by Jellyfrog** [comment]
@kvedder-amplex see failed tests https://github.com/librenms/librenms/pull/12101/checks?check_run_id=1639557328

**Comment by Jellyfrog** [comment]
Thank you for your contribution but we're closing this due to inactivity.  
Feel free to reopen if you plan on finishing it!

**Pull request closed without merging**